fix: Render tag description with <OAMarkdown>

alexchexes opened the pull request:
Similar to #272, this makes descriptions taken from the top-level `tags` list — where Markdown usage [is allowed by OAS](https://spec.openapis.org/oas/latest.html#tag-object) — to be rendered with `<OAMarkdown>` instead of plain text.

Before:

<img width="1572" height="522" alt="2025-07-26--12-02-37--352" src="https://github.com/user-attachments/assets/3539ab7e-302d-498e-bf30-d3eef60e82b9" />


After:

<img width="1572" height="522" alt="2025-07-26--12-02-28--054" src="https://github.com/user-attachments/assets/4d2ac955-ff60-4b4c-98ab-bd2e5d253029" />

alexchexes commented on src/components/Path/OAPathsGroup.vue:
```diff
@@ -54,9 +55,7 @@ function onPathClick(hash: string) {
-        <div v-if="hasDescription">
-          <p>{{ group.description }}</p>
-        </div>
+        <OAMarkdown v-if="hasDescription" :content="group.description" />
```
Actually, this is not the only inconsistency in this template regarding `props.` usage inside `<template>`. See lines 49 and 51 vs. 61 and 86, for example.
We probably need to adjust that and remove `props.` in all places in `<template>` since the Vue docs explicitly allow omitting it for clearer code.

enzonotario replied:
I prefer to keep the `props.` prefix to easily know it's a prop, and not a custom var/computed, etc. Also it's better for auto-completion.

However I know it's not consistent through the project. When I see a component without `props.` prefix, I try to refactor it to add the `props.` prefix

enzonotario commented on docs/public/openapi.json:
```diff
@@ -31,7 +31,7 @@
     },
     {
       "name": "Artists",
-      "description": "Everything about Argentine Rock artists"
+      "description": "Everything about `Argentine Rock artists` and their contributions to music history.\n\n> **Note:** You will need to authenticate yourself to access some of these endpoints using the [](/operations/loginUser) endpoint."
     }
```
no, it renders OAOperationLink via markdown-it plugin

enzonotario commented:
Nice @alexchexes , I just sent [a commit](https://github.com/enzonotario/vitepress-openapi/pull/281/commits/ed2cd20ea2708cb006e5efe0c783448b30b9ad79) to remove the `[&>p]:!my-0` class, since we don't use it in others `OAMarkdown` cases. I think you have setted it to better align the UI, however I think it's better to be consistent across multiple uses.

So, to allow to customize it on Project side, I've added the `OAMarkdown` class to easily target it via CSS, like:

```css
.OAMarkdown p {
  @apply my-0 !important;
}
```

We could add more specific targets in the future, to allow to customize just Tags Descriptions, or Reponses Descriptions, etc.

Please let me know what do you think

alexchexes commented:
> I think you have setted it to better align the UI, however I think it's better to be consistent across multiple uses.

Hi @enzonotario! I added that to keep the spacing exactly as it was before adding `<OAMarkdown>`, because there is `.oa-doc p` styling that adds these 0.25rem margins:

<img width="1704" height="1064" alt="chrome_2025-07-27--01-10-33--024" src="https://github.com/user-attachments/assets/0f4c6666-ec18-4e84-9c48-2f27bd1ce91f" />

So the E2E tests failed, and I added that "reset". But no problem — if you think this additional margin is OK even when there's only one `<p>`, then keep it 👌

enzonotario commented:
Ok, nice. Yes, previously the `p` didn't have that `0.25rem` margin. But now that we allow Markdown, I think it's better to fall back to default VitePress Markdown styling.

Thanks!
